content/geometry/CircleIntersection.h

```diff
@@ -1,25 +1,22 @@
 /**
- * Author: Simon Lindholm
+ * Author: Simon Lindholm, Victor Lecomte, chilli
  * Date: 2015-09-01
  * License: CC0
- * Description: Computes a pair of points at which two circles intersect. Returns false in case of no intersection.
+ * Description: Computes a pair of points at which two circles intersect. Returns number of intersections. If intersection is only at one point then that point is returned twice.
  * Status: somewhat tested
  */
 #pragma once
 
 #include "Point.h"
 
 typedef Point<double> P;
-bool circleIntersection(P a, P b, double r1, double r2,
-		pair<P, P>* out) {
-	P delta = b - a;
-	assert(delta.x || delta.y || r1 != r2);
-	if (!delta.x && !delta.y) return false;
-	double r = r1 + r2, d2 = delta.dist2();
+int circleInter(P a,P b,double r1,double r2,pair<P, P>& out){
+	P delta = b - a; double d2 = delta.dist2();
```

Comment:
move this down, possibly eliminate delta or reuse b for it; declarations of different types on the same line is a bit ugly, and the double declarations could be combined